install_files/securedrop-config/DEBIAN/postinst

```diff
@@ -13,6 +13,12 @@ remove_2fa_tty_req() {
     # Lets prevent this from bombing out the install though if it fails
     auth_file=/etc/pam.d/common-auth
     sed -i "/^auth\ required\ pam_google.*/d" ${auth_file} || true
+    # Since we are removing the pam_google module, we must ensure password
+    # authentication for sshd is disabled to ensure only key authentication is used.
+    grep -qF "PasswordAuthentication no" /etc/ssh/sshd_config || echo "PasswordAuthentication no" >> /etc/ssh/sshd_config
+    sed -i "/^UsePAM\ /s/\ .*/\ no/" /etc/ssh/sshd_config
+    sed -i "/^ChallengeResponseAuthentication\ /s/\ .*/\ no/" /etc/ssh/sshd_config
+    service ssh restart
```

Comment:
sorry, just noticed this - is using service ssh restart in postinst the right practice here? https://www.debian.org/doc/debian-policy/ch-opersys.html#running-initscripts says we should be using invoke-rc.d

(note this did work for me as expected during testing, just wondering if we should change it to the recommendation)

Reply:
Good point @redshiftzero , thanks for the link. From my understanding of that document, the intent of the guideline is for packagers to be mindful of other services and dependencies of these services on the host, to ensure all services change state graciously. Given that the state of services on SecureDrop instances is known (all SD instances should have the same services running), this seems less important in this context, but happy to make the change. Note that we do already use this method for apache: https://github.com/freedomofpress/securedrop/blob/develop/install_files/securedrop-app-code/DEBIAN/postinst#L119

Reply:
ok thanks for the explanation, since we (~)know the state of running SecureDrops, let's leave as is